### DISCUSSION TITLE
@conditional for checking whether an @load will succeed

### DISCUSSION THREAD

**Discussion by vpax** [zeek/zeek#3594] [General]
I'm maintaining some complex content that can optionally leverage a separate package if it's available. Currently it just `@load`s that package, forcing me to install it. I'd like a way to conditionally see whether the package is installed. I can then make decisions based on the result, such as `@load`ing it if it is, and `@load`ing an alternative if not, or whatever. AFAIK, there's no way to currently do this from within a script - seems like a handy, and pretty straightforward, addition.

**Comment by awelzel** [comment]
> AFAIK, there's no way to currently do this from within a script.

I've hacked this previously with absolute paths - usually to the dismay of others that needed to make it then work in another environment.
```
@if ( file_size("/abs/path/to/the-package/__load__.zeek") >= 0 )
@load the-package
@endif
```

Wonder if `file_exists()` and `absolute_load_path(path: string)` could be the primitives to make this a bit nicer. Can then build something like `package_exists()` or `can_load(p: string)` using these, wrapped in `@if`.

Main question could be what to return for `absolute_load_path("./consts.zeek")`, but maybe that's just not allowed (no relative paths and no absolute paths as input, making it clear ZEEKPATH is searched).

**Reply by JustinAzoff** [reply]
This does seem nice:

```
@if ( can_load("the-package") )
@load the-package
@endif
```

A while ago we did have the discussion about loading directories or globs directly, and not needing a `__load__` script.  This can also be used to solve similar problems of finding things dynamically.  Maybe not so much if packages are completely unrelated, more for the use case of 'plugin' like things like intel/seen.

**Comment by awelzel** [comment]
`can_load(...)` and `find_in_zeekpath()` will be available with Zeek 7.0.
